From 610b0a20f482e4011964b4935b5f14b344c9bac8 Mon Sep 17 00:00:00 2001
From: Seth Forshee <seth.forshee@canonical.com>
Date: Thu, 19 Jun 2014 16:12:36 -0500
Subject: [PATCH] trace-cmd: Add missing mode to open() call

touch_file() calls open(2) with O_CREAT but omits the required
mode argument. Fix this omission.

Signed-off-by: Seth Forshee <seth.forshee@canonical.com>
---
 trace-record.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/trace-record.c b/trace-record.c
index d3dd0c5..8438388 100644
--- a/trace-record.c
+++ b/trace-record.c
@@ -1995,7 +1995,7 @@ static void touch_file(const char *file)
 {
 	int fd;
 
-	fd = open(file, O_WRONLY | O_CREAT | O_TRUNC);
+	fd = open(file, O_WRONLY | O_CREAT | O_TRUNC, 0644);
 	if (fd < 0)
 		die("could not create file %s\n", file);
 	close(fd);
-- 
1.9.1

